src/libmongoc/src/mongoc/mongoc-client-side-encryption.c

```diff
@@ -1281,8 +1323,13 @@ _mongoc_cse_client_enable_auto_encryption (mongoc_client_t *client,
       GOTO (fail);
    }
 
-   client->topology->crypt = _mongoc_crypt_new (
-      opts->kms_providers, opts->schema_map, opts->tls_opts, error);
+   client->topology->crypt =
+      _mongoc_crypt_new (opts->kms_providers,
+                         opts->schema_map,
+                         opts->tls_opts,
+                         client->topology->csfle_override_path,
+                         client->topology->csfle_required,
+                         error);
```

Comment:
`opts->bypass_auto_encryption` may also need to be passed to disable loading `csfle`. That is new behavior added in DRIVERS-2287.